src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -629,7 +653,28 @@ private[fastq] object FastqDemultiplexer {
                          comment: Option[String],
                          originalBases: Option[String] = None,
                          originalQuals: Option[String] = None,
-                         readInfo: Option[ReadInfo] = None)
+                         readInfo: Option[ReadInfo] = None,
+                         numQ30Above: Int = 0) {
+
+    /** Masks bases that have a quality score less than or equal to a specified threshold.
+      * @param threshold The threshold for masking bases, exlusive. Bases with a quality score < threshold will be masked
+      * @param qualityEncoding The encoding used for quality scores in the Fastq file.
+      * @return a new DemuxRecord with updated bases
+      */
+    def maskLowQualityBases(threshold: PhredScore, qualityEncoding: QualityEncoding, metricsThreshold: PhredScore): DemuxRecord = {
+
+      val quals = this.quals.toCharArray
+      val numQ30Above = quals.map(q => if (q < metricsThreshold) 0 else 1).sum
+
+      if (threshold <= 0 || this.quals.forall(_ >= threshold)) {
+        this.copy(numQ30Above=numQ30Above)
+      } else {
+        val bases = this.bases.toCharArray
+        for (i <- Range(0, this.quals.length)) if (quals.charAt(i) < threshold) bases(i) = 'N'
+        this.copy(bases = bases.mkString, numQ30Above=numQ30Above)
```

Comment:
Rather that mutating the original array it is more "scala-y" to map and return a new array.
```suggestion
        val maskedBases = bases.zip(quals).map((base, qual) => if (qual < threshold) 'N' else base)
        this.copy(bases = maskedBases.mkString, numQ30Above=numQ30Above)
```